app/controllers/ticket_transactions_controller.rb

```diff
@@ -0,0 +1,92 @@
+class TicketTransactionsController < ApplicationController
+  before_action :authenticate_user
+  before_action :set_ticketing, except: :index
+  protect_from_forgery except: [:hook]
+
+  def index
+    @ticket_transactions = TicketTransaction.pending(params[:id]).decorate
+  end
+
+  def checkout_ticket
```

Comment:
CRITICAL: the checkout_ticket action is too large. For code quality, we should refactor this.